### ISSUE TITLE
Pseudo-locales don't work in Blazor WebAssembly

### ISSUE THREAD

**Issue by uecasm** [dotnet/aspnetcore#40630]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

After creating and publishing resources for `qps-ploc` and setting the default thread locale to it, Blazor WebAssembly's `IStringLocalizer` will render the default language instead.  Other languages work as expected.

### Expected Behavior

The `qps-ploc` locale should work as well (as should the other pseudo locales).

### Steps To Reproduce

https://github.com/uecasm/pseudo-wasm

This is just the default `blazorwasm` template with localization added from the tutorial.

Edit `Program.cs` and comment in exactly one of the `culture` lines to see differences in behaviour.  With `es-ES` selected, it displays the Spanish translation as expected.  With either of the others, it displays the default resources.

In particular it does not show the `qps-ploc` translation when that is selected, even though the resources DLL is generated and is included in the `blazor.boot.json` manifest.

### .NET Version

6.0.200

### Anything else?

https://stackoverflow.com/questions/71376446/pseudo-localization-in-blazor-webassembly

Visual Studio Code 1.65.0

.NET SDK (reflecting any global.json):
 Version:   6.0.200
 Commit:    4c30de7899

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.200\

Host (useful for support):
  Version: 6.0.2
  Commit:  839cdfb0ec

.NET SDKs installed:
  2.1.511 [C:\Program Files\dotnet\sdk]
  6.0.200 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

**Comment by javiercn** [comment]
@uecasm thanks for contacting us.

* Do those pseudo-locales work in .NET?
* Does localization work in your app if you use a different locale?

**Comment by uecasm** [comment]
Yes, they both work perfectly fine in a .NET Framework or .NET 6 Console app.  And I already said that the Spanish locale is working fine in this project, it's only the pseudo locale that's not working.

**Comment by javiercn** [comment]
@uecasm thanks for the clarifications.

I think this might be a runtime issue then with regards to globalization.
